### PR TITLE
update abyss to be able to handle fastq.gz input

### PR DESCRIPTION
**Pull request by cat-bro**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [X] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

abyss takes fasta, fastq and fastq.gz input types but using fastq.gz causes jobs to fail with 
```
error: Expected either `>' or `@' or 11 fields
and saw `' and 1 fields near
�A`�a�dataset_8721401.dat̽M���$��-nw}�H֩�s2�H ���˴�4�ٌ�����othcc����e��"� "<�=���������������_���?��������㯗���?�-����wN����޿�K�w����ۿ�?�?�ܿ��=����2~A�Ǟ�[���g�;~,%~���ݿ�S2�m�w1�g$�����o��v���1-˲��R��k���ڶ�����ܶ�V|��ݿ��g�m�v石��?h[����?��_�G���/nm꿣��������Rl1��Z-��M-��S1+%����cyz���s��<����_9���G�?��?���#���G3"�!�_o�8�g�p�/�>�_����_���I_���3���z,�4��橇2-k����m�z�o���G�T=D�����:m��۶�[�V�����<ϭ�Җ�_����KkkmiVڷ��������]�?�z3��=�����a�S���:�Y����s�3�����i8u�G��?�/���&����w�/E�'�������H�c�����_=���Rr]��'37���tm<w8q�S�N]��~�N�����!�?����cݣ׿�O^��Z���x6ksYm���?�7c��s٣�{����?�f���=�A?�����0�%~��A�q�N���{���+�Co8҈*���M⿺��c�����=��_~.S-8�=v?r���n�cc�{/
make: *** [/mnt/tools-indices/tool_dependencies/_conda/envs/mulled-v1-f64a501adeebbb46386bd1e43566490774f43fd0d2a7df1a17265ae6ff08a5e8/bin/abyss-pe.Makefile:555: abyss-1.fa] Error 1
make: *** Deleting file 'abyss-1.fa'
```

I've made an update that symlinks all of the input files and names the links *.<fasta|fastq|fastq.gz> so that abyss will understand what they are.  It's fairly laborious because there are four input fields, each with an arbitrary number of datasets.  I'd love to find a better way of writing this.  Is there a built-in method for symlinking inputs with extensions?

### CODE REVIEW

**Comment by cat-bro** [comment]
I'm waiting for the CI run to finish before adding another commit to potentially fix this bug

File: tools/abyss/abyss-pe.xml [hunk 1]
```diff
@@ -203,8 +238,8 @@ k=$k
             <repeat name="lib_repeat">
                 <conditional name="reads">
                     <param name="reads_selector" value="paired" />
-                    <param name="reads1" ftype="fastqsanger" value="assembly_sample-R1.fastq.gz" />
-                    <param name="reads2" ftype="fastqsanger" value="assembly_sample-R2.fastq.gz" />
+                    <param name="reads1" ftype="fastq.gz" value="assembly_sample-R1.fastq.gz" />
```
Comment (bernt-matthias) [inline tools/abyss/abyss-pe.xml:241]: Can you use `fastqsanger.gz` as ftype in the test? 

Guess this implies even more logic .. probably `$...ext.startswith("fastq")` might help.

**Comment by bernt-matthias** [comment]
Btw. what did I miss here: https://github.com/galaxyproject/tools-iuc/pull/3543 ?

**Comment by bgruening** [comment]
ping @cat-bro

**Comment by cat-bro** [comment]
@bernt-matthias in #3543 the test-data files were changed to .gz files but because the ftype was still 'fastqsanger' unzipped, the files were being unzipped upon upload.

File: tools/abyss/abyss-pe.xml [hunk 2]
```diff
@@ -215,15 +243,15 @@ k=$k
             <output name="stats" file="abyss-stats1.tab" />
         </test>
         <test>
-            <param name="se_reads" ftype="fastqsanger" value="assembly_sample-R1.fastq.gz,assembly_sample-R2.fastq.gz" />
+            <param name="se_reads" ftype="fastq" value="assembly_sample-R1.fastq.gz,assembly_sample-R2.fastq.gz" />
```
Comment (bernt-matthias) [inline tools/abyss/abyss-pe.xml:246]: I think ftype should also be fastq.gz / fastqsanger.gz

File: tools/abyss/abyss-pe.xml [hunk 3]
```diff
@@ -38,6 +39,49 @@
     </stdio>
     <version_command>ABYSS --version | head -n 1</version_command>
     <command><![CDATA[
+
+#def get_link_fname($data, $prefix, $index)
+    ## Make a name for symlinked files including extension
+    #return ''.join([str($prefix), '_', str($index), '.', str($data.ext)])
```
Comment (bernt-matthias) [inline tools/abyss/abyss-pe.xml:45]: So, for fastqsanger.gz input the extension will be fastqsanger.gz, or? Can abyss deal with that, or is it just important that the gz eztension is there?

Comment (cat-bro) [reply tools/abyss/abyss-pe.xml:45]: I believe that the only thing that matters is whether the .gz extension is there or not

File: tools/abyss/abyss-pe.xml (hunk 2 above)
Comment (cat-bro) [reply tools/abyss/abyss-pe.xml]: updated to fastq.gz

**Comment by bgruening** [comment]
Thanks!

**Comment by cat-bro** [comment]
thanks for reviewing @bernt-matthias and @bgruening!

**Comment by bgruening** [comment]
I hope its fixed now. I just have rerun it.

**Comment by cat-bro** [comment]
This update of abyss is still listed as an Invalid Tool in the toolshed.  The problem may have been fixed by https://github.com/galaxyproject/galaxy/pull/13202 merged a few days ago.

**Comment by nsoranzo** [comment]
@cat-bro I've reset the repository metadata on the ToolShed, does that help?

**Comment by cat-bro** [comment]
@nsoranzo yes, thank you!  All fixed.

**Pull request merged**